### PR TITLE
adding files and code

### PR DESCRIPTION
**Pull request by sameh93**
adding files and code

### CODE REVIEW

**Comment by LouisAmon** [comment]
I think a bit of cleanup is in order :)
Could you please put all the network-related code in `network.py` and the storage-related in `storage.py` ?

I suggest you put the AutoML-related stuff in `automl.py`

I don't think it makes sense to have an `iam.py` : it's better to declare the policies & roles directly next to the resources they are related to (that's how Pulumi does it in their examples too)

Also regarding the aesthetics : could you please correct the wildcard (*) imports and only import the modules actually used ?

It's a best practice in order not to pollute the namespace :)

**Comment by LouisAmon** [comment]
I suggest you put the compute-related stuff (your `lambdafnc.py`) in `compute.py`, so that there's no confusion with `lambda_function.py`
Also, please remove `handler.py` as you chose to name it `lambda_function.py` which is great!

File: __main__.py [hunk 1]
```diff
@@ -14,3 +14,17 @@
...
+import vpc
+import vpn
+import kms
+import s3
+import vpcs3endpoint
+import iam
+import lambdafnc
+
+
+print("starting the proccess...")
```
Comment (LouisAmon) [inline __main__.py]: Let's avoid print statements if they aren't providing necessary information (I don't want to pollute stdout)

File: iam.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import json
+import pulumi
+from pulumi_aws import iam
+
+
+# Create an AWS resource (IAM policy that grants Sagemaker access on S3 bucket)
+policy = iam.Policy("AmazonSageMaker-ExecutionPolicy",
```
Comment (LouisAmon) [inline iam.py]: Could you please re-indent this section to be as leftmost as possible ?

File: iam.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                                            {
+                                                "Effect": "Allow",
+                                                "Action": [
+                                                    "s3:GetObject",
+                                                    "s3:PutObject",
+                                                    "s3:DeleteObject",
+                                                    "s3:ListBucket"
+                                                ],
+                                                "Resource": [
+                                                    "arn:aws:s3:::*"
```
Comment (LouisAmon) [inline iam.py]: Can we restrict to the S3 bucket we created ?

File: s3.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import json
+import pulumi
+from pulumi_aws import s3, kms
+from kms import *
+
+
+#Inputs 
+currentregion = pulumi.Config('aws')
```
Comment (LouisAmon) [inline s3.py]: Can you please name the config `aws_config` and put the `account_id` variable within the same namespace ? (`aws:account_id`)

File: s3.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+current_region = currentregion.require('region')
+print("this is the print region:",currentregion.require('region'))
+
+currentaccountnumber = pulumi.Config('account')
+current_account_number = currentaccountnumber.require('id')
+print("this is the print account number:",currentaccountnumber.require('id'))
```
Comment (LouisAmon) [inline s3.py]: Print statements :)

File: vpc.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import json
+import pulumi
+from pulumi_aws import ec2
+
+# Create an AWS resource (VPC)
+vpc = ec2.Vpc(
+            "custom-VPC",
+            cidr_block="10.10.0.0/16",
```
Comment (LouisAmon) [inline vpc.py]: Should we hard-code this ?

File: vpcs3endpoint.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import json
+import pulumi
+from pulumi_aws import ec2
+from vpc import *
+
+# Create an AWS resource (VPC endpoint with S3)
+s3endpoint = ec2.VpcEndpoint("s3", 
+      service_name= "com.amazonaws.eu-west-1.s3",
```
Comment (LouisAmon) [inline vpcs3endpoint.py]: Can we get the region from the config ?

File: vpn.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+#Create an AWS resource (vitrtual private gateway)
+vpnGateway = ec2.VpnGateway("vpnGateway", 
+     vpc_id= vpc.id,
+     tags={
+        "Name": "vpnGateway",
+        })
+# Create an AWS resource (customer gateway)
+customer_gateway = ec2.CustomerGateway("CustomerGateway", 
+    bgp_asn=65000,
```
Comment (LouisAmon) [inline vpn.py]: 65000 ? Why is this hard-coded ?

File: vpn.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        })
+# Create an AWS resource (customer gateway)
+customer_gateway = ec2.CustomerGateway("CustomerGateway", 
+    bgp_asn=65000, 
+    ip_address= vpnip,
+    type="ipsec.1",
+    tags={
+        "Name": "CustomerGateway",
+    })
+'''
```
Comment (LouisAmon) [inline vpn.py]: Cleanup dev comment ? Or is it there for a reason ?

File: __main__.py (hunk 1 above)
Comment (sameh93) [reply __main__.py]: done

File: iam.py (hunk 3 above)
Comment (sameh93) [reply iam.py]: done

File: s3.py (hunk 4 above)
Comment (sameh93) [reply s3.py]: done

File: s3.py (hunk 5 above)
Comment (sameh93) [reply s3.py]: done

File: vpc.py (hunk 6 above)
Comment (sameh93) [reply vpc.py]: done, took from input

File: vpn.py (hunk 8 above)
Comment (sameh93) [reply vpn.py]: it is a fixed number

File: vpn.py (hunk 9 above)
Comment (sameh93) [reply vpn.py]: we will discuss this section on slack as i comment it for a reason

File: vpcs3endpoint.py (hunk 7 above)
Comment (sameh93) [reply vpcs3endpoint.py]: done

**Pull request closed without merging**